nucypher/blockchain/eth/sol/source/contracts/WorkLock.sol

```diff
@@ -89,94 +119,370 @@ contract WorkLock {
     }
 
     /**
     * @notice Calculate amount of ETH that will be refund for completing specified amount of work
     * @dev This value will be fixed only after end of bidding
     */
-    function workToETH(uint256 _completedWork) public view returns (uint256) {
-        return _completedWork.mul(ethSupply).mul(boostingRefund).div(tokenSupply.mul(SLOWING_REFUND));
+    function workToETH(uint256 _completedWork, uint256 _depositedETH) public view returns (uint256) {
+        uint256 baseETHSupply = bidders.length * minAllowedBid;
+        // when all participants bid with the same minimum amount of eth
+        if (bonusETHSupply == 0) {
+            return workToETH(_completedWork, baseETHSupply, tokenSupply);
+        }
+
+        uint256 bonusWork = 0;
+        uint256 bonusETH = 0;
+        uint256 baseTokenSupply = bidders.length * minAllowableLockedTokens;
+
+        if (_depositedETH > minAllowedBid) {
+            bonusETH = _depositedETH - minAllowedBid;
+            uint256 bonusTokenSupply = tokenSupply - baseTokenSupply;
+            bonusWork = ethToWork(bonusETH, bonusTokenSupply, bonusETHSupply);
+
+            if (_completedWork <= bonusWork) {
+                return workToETH(_completedWork, bonusETHSupply, bonusTokenSupply);
+            }
+        }
+
+        _completedWork -= bonusWork;
+        return bonusETH + workToETH(_completedWork, baseETHSupply, baseTokenSupply);
     }
 
     /**
     * @notice Get remaining work to full refund
     */
-    function getRemainingWork(address _depositor) external view returns (uint256) {
-        WorkInfo storage info = workInfo[_depositor];
-        uint256 completedWork = escrow.getCompletedWork(_depositor).sub(info.completedWork);
+    function getRemainingWork(address _bidder) external view returns (uint256) {
+        WorkInfo storage info = workInfo[_bidder];
+        uint256 completedWork = escrow.getCompletedWork(_bidder).sub(info.completedWork);
         uint256 remainingWork = ethToWork(info.depositedETH);
         if (remainingWork <= completedWork) {
             return 0;
         }
-        return remainingWork.sub(completedWork);
+        return remainingWork - completedWork;
+    }
+
+    /**
+    * @notice Get length of bidders array
+    */
+    function getBiddersLength() external view returns (uint256) {
+        return bidders.length;
     }
 
     /**
     * @notice Bid for tokens by transferring ETH
     */
     function bid() external payable {
-        require(block.timestamp >= startBidDate && block.timestamp <= endBidDate,
-            "Bid is open during a certain period");
+        require(block.timestamp >= startBidDate, "Bidding is not open yet");
+        require(block.timestamp < endBidDate, "Bidding is already finished");
         WorkInfo storage info = workInfo[msg.sender];
+
+        // first bid
+        if (info.depositedETH == 0) {
+            require(msg.value >= minAllowedBid, "Bid must be at least minimum");
```

Comment:
Some bidders will see min bid and will try with that value which will result in an error. And it will be need to explain why need to put bid 10 ETH + 1 wei, so I'm for `>=`. But want to hear other opinions

Reply:
I agree with @vzotova , it's a weird UX to state the minimum as 10 ETH but then force refund bids of that size. It's very unlikely that not a single bidder exceeds the minAllowedBid anyway